File: snippets/java/M09_PassengersProgram.java
import inout.In;
import inout.Out;

class Passenger {

    final String firstName;
    final String lastName;

    Passenger(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    void display() {
        Out.println(firstName + " " + lastName);
    }
}

class PassengersProgram {

    static Passenger createPassengerFromInput() {
        Out.print("Enter first name: ");
        String firstName = In.readLine();
        Out.print("Enter last name: ");
        String lastName = In.readLine();

        return new Passenger(firstName, lastName);
    }

    public static void main(String[] args) {
        Passenger lisa = new Passenger("Lisa", "Ha");
        Passenger user = createPassengerFromInput();

        lisa.display();
        user.display();
    }
}

File: snippets/python/m09_passengers_program.py
class Passenger:
    def __init__(self, first_name, last_name):
        self.first_name = first_name
        self.last_name = last_name

    def display(self):
        print(self.first_name + " " + self.last_name)


def create_passenger_from_input():
    first_name = input("Enter first name: ")
    last_name = input("Enter last name: ")

    return Passenger(first_name, last_name)


if __name__ == "__main__":
    lisa = Passenger("Lisa", "Ha")
    user = create_passenger_from_input()

    lisa.display()
    user.display()
